### ISSUE TITLE
[MPLUGIN-39] adding package-info.java breaks build

### ISSUE THREAD

**Issue by jira-importer** [apache/maven-plugin-tools#495]
**[Florian Kleedorfer](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=phlow)** opened **[MPLUGIN-39](https://issues.apache.org/jira/browse/MPLUGIN-39?redirect=false)** and commented

As soon as I add a package-info.java file to the package of my mojo, the build breaks with the following output:

```
[INFO] [plugin:descriptor]
[INFO] Using 2 extractors.
[INFO] Applying extractor for language: java
[INFO] ------------------------------------------------------------------------
[ERROR] FATAL ERROR
[INFO] ------------------------------------------------------------------------
[INFO] 0
[INFO] ------------------------------------------------------------------------
[DEBUG] Trace
java.lang.ArrayIndexOutOfBoundsException: 0
        at org.apache.maven.tools.plugin.extractor.java.JavaMojoDescriptorExtractor.getJavaClass(JavaMojoDescriptorExtractor.java:534)
        at org.apache.maven.tools.plugin.extractor.java.JavaMojoDescriptorExtractor.execute(JavaMojoDescriptorExtractor.java:553)
        at org.apache.maven.tools.plugin.scanner.DefaultMojoScanner.populatePluginDescriptor(DefaultMojoScanner.java:84)
        at org.apache.maven.plugin.plugin.AbstractGeneratorMojo.execute(AbstractGeneratorMojo.java:135)
        at org.apache.maven.plugin.DefaultPluginManager.executeMojo(DefaultPluginManager.java:443)
        at org.apache.maven.lifecycle.DefaultLifecycleExecutor.executeGoals(DefaultLifecycleExecutor.java:539)
        at org.apache.maven.lifecycle.DefaultLifecycleExecutor.executeStandaloneGoal(DefaultLifecycleExecutor.java:493)
        at org.apache.maven.lifecycle.DefaultLifecycleExecutor.executeGoal(DefaultLifecycleExecutor.java:463)
        at org.apache.maven.lifecycle.DefaultLifecycleExecutor.executeGoalAndHandleFailures(DefaultLifecycleExecutor.java:311)
        at org.apache.maven.lifecycle.DefaultLifecycleExecutor.executeTaskSegments(DefaultLifecycleExecutor.java:278)
        at org.apache.maven.lifecycle.DefaultLifecycleExecutor.execute(DefaultLifecycleExecutor.java:143)
        at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:334)
        at org.apache.maven.DefaultMaven.execute(DefaultMaven.java:125)
        at org.apache.maven.cli.MavenCli.main(MavenCli.java:280)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:39)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:25)
        at java.lang.reflect.Method.invoke(Method.java:585)
        at org.codehaus.classworlds.Launcher.launchEnhanced(Launcher.java:315)
        at org.codehaus.classworlds.Launcher.launch(Launcher.java:255)
        at org.codehaus.classworlds.Launcher.mainWithExitCode(Launcher.java:430)
        at org.codehaus.classworlds.Launcher.main(Launcher.java:375)
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 2 seconds
[INFO] Finished at: Wed Oct 17 11:48:52 CEST 2007
[INFO] Final Memory: 4M/1016M
[INFO] ------------------------------------------------------------------------
```

---

**Attachments:**
- [MPLUGIN-39.patch](https://issues.apache.org/jira/secure/attachment/12717502/MPLUGIN-39.patch) (_548 bytes_)

**Sub-tasks:**
- [MPLUGIN-66](https://issues.apache.org/jira/browse/MPLUGIN-66) Add unit test

1 votes, 0 watchers

**Comment by jira-importer** [comment]
**[Benjamin Bentmann](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=bentmann)** commented

I can confirm this bug for maven-plugin-tools-java-2.1. However, the bug seems to have been fixed already in SVN (by applying MNG-2930.diff): I built from the current trunk and it work.

The attached patch adds a package-info.java to one of the existing unit tests and should serve as a regression test for this issue.
